terraform/modules/rds/database.tf

```diff
@@ -0,0 +1,34 @@
+#tfsec:ignore:aws-rds-specify-backup-retention:TODO CLDC-2261
+#tfsec:ignore:aws-rds-enable-performance-insights:TODO CLDC-2260 if necessary
+#tfsec:ignore:AVD-AWS-0176:iam authentication not suitable as tokens only last 15minutes, password authentication preferred
+resource "aws_db_instance" "main" {
+  #checkov:skip=CKV_AWS_129:cloudwatch logs TODO CLDC-2260
+  #checkov:skip=CKV_AWS_118:monitoring TODO CLDC-2260
+  #checkov:skip=CKV_AWS_161:iam authentication not suitable as tokens only last 15minutes, password authentication preferred
+  #checkov:skip=CKV_AWS_353:performance insights TODO CLDC-2260 if necessary
+  #checkov:skip=CKV_AWS_354:performance insights TODO CLDC-2260 if insights are necessary
+  #checkov:skip=CKV2_AWS_30:query logging TODO CLDC-2260
+  #checkov:skip=CKV2_AWS_60:copy tags to snapshots TODO CLDC-2261
+  identifier                 = "${var.prefix}-postgres-db"
+  apply_immediately          = true
+  auto_minor_version_upgrade = true
+  allocated_storage          = var.allocated_storage #units are GiB
+  db_name                    = "data_collector"
+  db_subnet_group_name       = var.db_subnet_group_name
+  deletion_protection        = true # needs to be set to false and applied if you need to delete the DB
+  engine                     = "postgres"
+  engine_version             = "13.11"
```

Comment:
I think we're using Postgres 13.5 (looking at the existing pipelines etc). Closest version we can get is 13.7 on AWS, but I've taken 13.11 here as I assume the latest minor version will be the best while compatible